src/arrayMethodJoin.js

```diff
@@ -4,8 +4,32 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let newSeparator = separator;
+    const newArray = [];
+    let string = '';
+
+    if (!separator) {
+      newSeparator = `${separator}`;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        newArray.push('');
+      } else {
+        newArray.push(this[i]);
+      }
+    }
+
+    for (let i = 0; i < newArray.length; i++) {
+      if (i === newArray.length - 1) {
+        string += newArray[i];
+      } else {
+        string += newArray[i] + newSeparator;
+      }
+    }
```

Comment:
We should not use a loop twice with a buffer variable like new Array. 
Just inside your first loop make checks

```
if () {
  string += this[i];
}
if (...) {
  string += separator;
}
```

This is all that you need)

Reply:
Well, newArray is needed for this test:
 test(`Should work when array has null and undefined`, () => {
  expect([false, null, 1, undefined, 0, NaN, ''].join2(','))
    .toBe('false,,1,,0,NaN,');
});

It filters undifned and null from array.

Reply:
Got you, but I can`t resolve problem with null and undefined. Here is my code:
![image](https://user-images.githubusercontent.com/62021566/209243333-10312082-0f6c-41ac-bd9f-bf86ac644d6c.png)
Unfortunately, I can`t pass tests, because this condition is not working for some reason.

Reply:
So i meant just use your conditions at if statement
```
if (this[i] === null || this[i] === undefined) {
  string += this[i];
}
if (i === newArray.length - 1) {
  string += separator;
}
```